types/express-serve-static-core/index.d.ts

```diff
@@ -71,8 +71,10 @@ export interface IRouterMatcher<T, Method extends 'all' | 'get' | 'post' | 'put'
 }
 
 export interface IRouterHandler<T> {
-    (...handlers: RequestHandler[]): T;
-    (...handlers: RequestHandlerParams[]): T;
+  // tslint:disable-next-line no-unnecessary-generics (This generic is meant to be passed explicitly.)
+  <P = ParamsDictionary, ResBody = any, ReqBody = any, ReqQuery = ParsedQs>(...handlers: Array<RequestHandler<P, ResBody, ReqBody, ReqQuery>>): T;
+  // tslint:disable-next-line no-unnecessary-generics (This generic is meant to be passed explicitly.)
+  <P = ParamsDictionary, ResBody = any, ReqBody = any, ReqQuery = ParsedQs>(...handlers: Array<RequestHandlerParams<P, ResBody, ReqBody, ReqQuery>>): T;
```

Comment:
This change broke `@types/morgan`, I’m fixing that in <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50065>.

---

Also, this should be indented and formatted correctly:
```suggestion
    // tslint:disable-next-line no-unnecessary-generics (This generic is meant to be passed explicitly.)
    <P = ParamsDictionary, ResBody = any, ReqBody = any, ReqQuery = ParsedQs>(
        ...handlers: Array<RequestHandler<P, ResBody, ReqBody, ReqQuery>>
    ): T;
    // tslint:disable-next-line no-unnecessary-generics (This generic is meant to be passed explicitly.)
    <P = ParamsDictionary, ResBody = any, ReqBody = any, ReqQuery = ParsedQs>(
        ...handlers: Array<RequestHandlerParams<P, ResBody, ReqBody, ReqQuery>>
    ): T;
```

Reply:
Ah okay, my mistake. 

I've raised a PR that should fix this issue. It allows both not passing generics and passing generics. Hopefully this will mean downstream types and packages won't need an update.